FE-Initial Frontend App Setup

viriponce opened the pull request:
This PR addresses Issue #69.

Most of the files involved in this PR were automatically created during the setting up a React application using Vite.

For containerizing the application with Docker the following configuration files were created:
- frontend/.dockerignore
- frontend/Dockerfile
- frontend/docker-compose.yml

The following file was created to specify files that Git should ignore:

- frontend/.gitignore

MarisiaS commented on frontend/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM node:21.6.2
+
```
Add the line `ENV CHOKIDAR_USEPOLLING=true` to the Dockerfile to ensure that changes in the code are automatically reflected in the container when using Docker on Windows.

viriponce commented:
@MarisiaS 
Added the ENV command.
I deleted the docker container and rebuilt it. It's working on mac as it did before, the changes in the code are reflected in the browser when saved it.

MarisiaS reviewed (approved):
LGTM

Thanks Viri!
